# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/vr/ipc/VRProcessManager.h
# Commit: 9c0c9e80f309
# Full Hash: 9c0c9e80f30955b6fdb3a2995ac7d7299ef30342
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-05-23 15:52:15
# Regressor Bug: 1550422
# File Overlap Count: 2
# Description:
#   Bug 1550422 - P9. Sync preferences in VR process when they change. r=daoshengmu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D31017
# ==============================================================================

diff -r 10c153ddbaea -r 9c0c9e80f309 gfx/vr/ipc/VRProcessManager.h
--- a/gfx/vr/ipc/VRProcessManager.h	Wed May 22 13:23:07 2019 +0000
+++ b/gfx/vr/ipc/VRProcessManager.h	Wed May 22 12:42:33 2019 +0000
@@ -53,6 +53,7 @@
   bool CreateGPUVRManager(base::ProcessId aOtherProcess,
                           mozilla::ipc::Endpoint<PVRGPUChild>* aOutEndpoint);
   void OnXPCOMShutdown();
+  void OnPreferenceChange(const char16_t* aData);
   void CleanShutdown();
   void DestroyProcess();
 
@@ -76,6 +77,10 @@
   RefPtr<Observer> mObserver;
   VRProcessParent* mProcess;
   VRChild* mVRChild;
+  // Collects any pref changes that occur during process launch (after
+  // the initial map is passed in command-line arguments) to be sent
+  // when the process can receive IPC messages.
+  nsTArray<mozilla::dom::Pref> mQueuedPrefs;
 };
 
 }  // namespace gfx
